src/list.cpp

```diff
@@ -16,22 +16,39 @@ namespace bpftrace {
 const std::string kprobe_path = "/sys/kernel/debug/tracing/available_filter_functions";
 const std::string tp_path = "/sys/kernel/debug/tracing/events";
 
-bool search_probe(const std::string &probe, const std::string search)
+std::string replace_all(const std::string &str, const std::string &from,
+			const std::string &to)
 {
-  std::string s = search;
-  char remove[] = "*.?";
-  unsigned int i;
+    std::string result(str);
+    std::string::size_type
+        index = 0,
+        from_len = strlen(from.c_str()),
```

Comment:
Sorry, just spotted one more thing - this can be just `from.size()` now

Reply:
you are right!.